group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordHelpers.java

```diff
@@ -408,6 +411,61 @@ public static CoordinatorRecord newConsumerGroupCurrentAssignmentTombstoneRecord
         );
     }
 
+    /**
+     * Creates a ConsumerGroupRegularExpression record.
+     *
+     * @param groupId   The consumer group id.
+     * @param regex     The regular expression.
+     * @param metadata  The metadata associated with the regular expression.
+     * @return The record.
+     */
+    public static CoordinatorRecord newConsumerGroupRegularExpressionRecord(
+        String groupId,
+        String regex,
+        ResolvedRegularExpression metadata
```

Comment:
nit: referring ResolvedRegularExpression as metadata throughout the code was a bit confusing at first glance. I can't think of a better name though 😅 maybe ResolvedRegexMetadata?

Reply:
Agreed. I renamed it to `resolvedRegularExpression`.